astroquery/esasky/tests/test_esasky_remote.py

```diff
@@ -1,352 +1,281 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 import os
-import shutil
-import tempfile
 
 import pytest
 from astropy.io.fits.hdu.hdulist import HDUList
 
-from astroquery import log
 from astroquery.utils.commons import TableList
 from astroquery.utils.tap.model.tapcolumn import TapColumn
 from astroquery.utils.tap.model.taptable import TapTableMeta
-from ... import esasky
-
-ESASkyClass = esasky.core.ESASkyClass()
+from astroquery.esasky import ESASky
```

Comment:
In most cases, I think it would even be better to use a class instance rather than directly call the method, we do that in the case for alma or cadc

Reply:
Tracking issue at #2528

Reply:
> It's nothing bad, as the `__call__` method is not defined for any of these classes, so it just returns a new instance from the existing instance.

Not quite, a new instance is returned because `__call__()` is implemented by `BaseQuery`:
https://github.com/astropy/astroquery/blob/404fa410820bc97a27fcfa18d1f10e9400651374/astroquery/query.py#L182-L184

Having instances that are named as if they are classes and can also create new instances as if they are classes but despite that are not classes can get very confusing, especially if coupled with the way much of `astroquery` implements configuration settings as class attributes. Consider the following:
```python
>>> from astroquery.esasky import ESASky, ESASkyClass
>>> ESASky.TIMEOUT
1000
>>> ESASky.TIMEOUT = 123
>>> ESASky().TIMEOUT
1000
>>> ESASkyClass.TIMEOUT = 987
>>> ESASkyClass().TIMEOUT
987
```
This is a design decision I cannot understand, but changing it would be a huge API update, and definitely far beyond the scope of this pull request.

Reply:
This design choice was made before my time in astroquery. I'm on a similar opinion that it would not be my preference, but at least I would be happy to see consistency in the docs, etc.